core/src/main/java/org/apache/iceberg/MetadataColumns.java

```diff
@@ -53,6 +53,8 @@ private MetadataColumns() {
   public static final String DELETE_FILE_ROW_FIELD_NAME = "row";
   public static final int DELETE_FILE_ROW_FIELD_ID = Integer.MAX_VALUE - 103;
   public static final String DELETE_FILE_ROW_DOC = "Deleted row values";
+  public static final int POSITION_DELETE_TABLE_PARTITION_FIELD_ID = Integer.MAX_VALUE - 104;
```

Comment:
It seems we have a static column in the metadata table that we plan to populate via the mechanism for metadata columns. It looks a little bit suspicious to me.

I feel we should pick one of these options:
- Have only `path`, `pos`, `row` columns in the table and use `_partition` and `_spec_id` metadata columns. That will mean we have to support filter pushdown on metadata columns. It is easy to handle this on the Spark side but we will also have to adapt ALL of our binding code to allow binding predicates with metadata columns. The last part will be a big change.
- Make `partition` and `spec_id` columns static and use `DataTask`.

I kind of like using `FileScanTask` for this effort to support vectorized reads so the first option seems preferable.

Thoughts, @szehon-ho @RussellSpitzer @rdblue?

Reply:
My concern with the marker `FileScanTask` is that if an engine is implementing metadata tables like normal reads, then we've introduced a correctness problem because it doesn't know to read the new task differently. I think the cleanest way is probably to use `DataTask`.

I think the concern about using `DataTask` is valid, since it exposes a row-based interface and isn't intended for large uses that benefit from vectorization. It was originally intended for small tables, like `snapshots`.

However, I've been thinking for a while that a significant improvement is to adapt Arrow record batches into rows, so we can take advantage of vectorized reads in all cases, not just when the engine supports a vectorized format. That is probably way faster, so we could explore doing that here and using a joined row.

Reply:
Thanks for taking a look.  It's getting a bit more messy than anticipated.

Was spending a lot of time yesterday looking at the marker FileScanTask approach and it does get messy.  Especially  trying to add another constant column (spec_id) to this table, not only is it another use of metadata column  to populate it, we have to figure out how to make a residual expression without the constant column as otherwise the default file read code tries to filter too aggressively if there is a spec_id filter (as the stats do not exist on the file and the pruning code skips it) .  In short , adding any constant columns to this table, gets a bit messy.

Makes sense to explore using some kind of StaticDataTask.  I think the problem for that, currently it would not do split() and implement any file residual filtering on position-delete row values if there are any filters there, and of course vectorization, and it'll be re-implementing these if we want it.  But maybe I am over-optimizing this and in the first cut we can live without those and add them later.  Can look at this approach, and see if I hit any major blockers.

Reply:
Hi guys, I hit the first issue , as StaticDataTask is in core module, there is no way currently for it to access Parquet, ORC, and the file readers.  So maybe engines, like Spark, can pass it in?  This would seem to require an API change in any case on DataTask.

So something like, adding a new method DataTask::rows(Function<DeleteFile, Schema, Expression> filter readerFunction)?  

This can also take care of encryption, which seems to also be a callback (for example DeleteFilter::inputFile(), which is implemented by Spark/Flink).

Reply:
I guess the only thing I can think to do is to provide a shared implementation of positionDeleteReader in data module, but each engine still needs to check the type of table (or FileScanTask) and then manually plug this in to the DataTask.

Reply:
cc @rdblue @aokolnychyi @rdblue appreciate any thoughts, I think its a good idea to try to make this table usable across engines using existing DataTask,  without extra if-else handling, but not sure we can do it as of today due to the dependency.  Will try to explore the above , but it  still will not be as clean

Reply:
Hey guys I really tried hard on this, but couldn't find a way.  We looked with @RussellSpitzer  and looks like Trino actually wont break if we had a new table (they support each one manually, and actually i think a good portion don't even use our MetadataTable class altogether).   Discussed a bit with @aokolnychyi as well and coudn't find a good way.  Made a latest pr with a least bad way, piggy backing off of some of the latest RowReader refactoring